python/semantic_kernel/connectors/ai/open_ai/services/open_ai_handler.py

```diff
@@ -149,6 +161,32 @@ def _create_model_args(
                 ]
             else:
                 model_args["functions"] = functions
+
+        if request_settings.data_source_settings is not None:
+            model_args["extra_body"] = {
```

Comment:
could you make this part of the data source settings class? and maybe you can then use the model_dump setting of pydantic, that way different versions of this (for AI search and for Cosmos/Mongo can work the same way more easily

Reply:
refactored the settings to have the highest level contain dataSources, and AI search contains type and parameters. I think this should be nicely extensible but let me know if you have further thoughts on it.